### PR TITLE
feat(teuto-portal-k8s-worker): initial implementation

### PR DESCRIPTION
**Pull request by tasches**
add helmchart for the teutoportal-worker

### CODE REVIEW

**Review by cwrau — CHANGES_REQUESTED** [review]
Just one nitpick, it's called the `teuto-portal` 😉

Please adjust accordingly

And you should do a rebase

**Comment by tasches** [comment]
> Just one nitpick, it's called the `teuto-portal` 😉
> 
> Please adjust accordingly
> 
> And you should do a rebase

done.

**Comment by cwrau** [comment]
> > Just one nitpick, it's called the `teuto-portal` 😉
> > 
> > Please adjust accordingly
> > 
> > And you should do a rebase
> 
> done.

I'm sorry, I was not clear, `teutoportal` should be `teuto-portal`. In total it's `teuto-portal-k8s-worker` 😅

File: charts/teuto-portal/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            httpGet:
+              path: /-/healthy
+              port: metrics
+          resources: {{- toYaml .Values.worker.resources | nindent 12 }}
+          volumeMounts:
+            - mountPath: /kubeconfig
+              name: clusterconfig
+              readOnly: true
+            - mountPath: /tmp
+              name: tmp
```
Comment (cwrau) [inline charts/teuto-portal/templates/deployment.yaml:62]: Did you check if this is necessary?

File: charts/teuto-portal/templates/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{{- if  not .Values.worker.existingSecret }}
```
Comment (cwrau) [inline charts/teuto-portal/templates/secret.yaml:1]: I'm thinking, are we ever going to not use the existing DB? I assume not 🤔

So I guess we can just use the existing secret and require it, just like the kubeconfig

File: charts/teuto-portal/values.schema.json [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+              "type": "string",
+              "pattern": "^(\\d+.*){1,3}$"
+            },
+            "loggingFormat": {
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        },
+        "resources": {
```
Comment (cwrau) [inline charts/teuto-portal/values.schema.json:196]: You can reference the real schema, see the base-cluster scheme

File: charts/teuto-portal/values.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      name: ""
+  config:
+    dryRun: false
+    singleRun: false
+    otelExporterTracesEndpoint: ""
+    syncInterval: 10s
+    watchK8sEvents: false
+    minimumCPUFlavour: 2
+    flavourPrefixFilter: ""
+    clusterNamespace: customer-clusters
```
Comment (cwrau) [inline charts/teuto-portal/values.yaml:42]: You should leave these empty to use the worker defaults

File: charts/teuto-portal/values.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    secret:
+      name: ""
+  config:
+    dryRun: false
+    singleRun: false
+    otelExporterTracesEndpoint: ""
+    syncInterval: 10s
+    watchK8sEvents: false
+    minimumCPUFlavour: 2
+    flavourPrefixFilter: ""
```
Comment (cwrau) [inline charts/teuto-portal/values.yaml:41]: This is a list of prefixes, in the environment variable joined by `, `

**Review by cwrau — CHANGES_REQUESTED** [review]

File: charts/teuto-portal/values.yaml (hunk 5 above)
Comment (tasches) [reply charts/teuto-portal/values.yaml:41]: so not a string instead something like this {[prefix1,prefix2,..] right?

File: charts/teuto-portal/values.yaml (hunk 4 above)
Comment (tasches) [reply charts/teuto-portal/values.yaml:42]: I can remove them entirely and just left them in the schema or did you mean for example 
`syncInterval: ""`

File: charts/teuto-portal/templates/secret.yaml (hunk 2 above)
Comment (tasches) [reply charts/teuto-portal/templates/secret.yaml:1]: removed

File: charts/teuto-portal/templates/_helpers.tpl [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{{- define "portalworker.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.worker.image "global" .Values.global) }}
+{{- end -}}
+
+{{- define "portalworker.jdbc" -}}
+{{- $portString := .Values.worker.database.port | int -}}
```
[the remark this answers is not in the document]
Comment (tasches) [reply charts/teuto-portal/templates/_helpers.tpl:6]: renamed

File: charts/teuto-portal/values.yaml (hunk 4 above)
Comment (cwrau) [reply charts/teuto-portal/values.yaml]: Setting them to an empty value for strings and the rest you can leave out and write a ci test file for them

Or you remove all of them and write a ci file for them

File: charts/teuto-portal/values.yaml (hunk 5 above)
Comment (cwrau) [reply charts/teuto-portal/values.yaml]: Yes, so by default `[]` and a test ci file with some entries

Comment (tasches) [reply charts/teuto-portal/values.yaml]: done

File: charts/teuto-portal/values.yaml (hunk 4 above)
Comment (tasches) [reply charts/teuto-portal/values.yaml]: done

File: charts/teuto-portal-k8s-worker/templates/_helpers.tpl [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{{- define "portalworker.image" -}}
+{{ include "common.images.image" (dict "imageRoot" .Values.worker.image "global" .Values.global) }}
+{{- end -}}
+
+{{- define "portalworker.jdbc" -}}
+{{- $port := .Values.worker.database.port | int -}}
+{{ printf "jdbc:postgresql://%s:%d/teuto_domain?currentSchema=app_public" .Values.worker.database.host $port }}
+{{- end -}}
+
+{{- define "portalworker.dbcredentials" -}}
```
Comment (cwrau) [inline charts/teuto-portal-k8s-worker/templates/_helpers.tpl:10]: I would rename this to something like `portalworker.dbCredentials.secretName` to be more explicit

File: charts/teuto-portal-k8s-worker/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+            - name: DB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ . }}
+                  key: password
+            {{- end }}
+            {{- range $k, $v := .Values.worker.config }}
+            {{- if or (typeIs "bool" $v) ($v) }}
+            - name: {{ regexReplaceAll `_(\d+)` ($k | snakecase | upper) "$1" }}
+              value: {{ $v | quote }}
```
Comment (cwrau) [inline charts/teuto-portal-k8s-worker/templates/deployment.yaml:49]: This doesn't work for the `flavourPrefixFilter`, it results in `"[teuto teuto-net]"` but it should be `"teuto,teuto-net"`.

I guess you should create an `if` with a `typeis "slice"` (I think) and join them.

Maybe just adding `| join ","` in the pipe before the quote could work 🤔

**Review by cwrau — CHANGES_REQUESTED** [review]

File: charts/teuto-portal-k8s-worker/templates/_helpers.tpl (hunk 7 above)
Comment (tasches) [reply charts/teuto-portal-k8s-worker/templates/_helpers.tpl]: renamed

File: charts/teuto-portal-k8s-worker/templates/deployment.yaml (hunk 8 above)
Comment (tasches) [reply charts/teuto-portal-k8s-worker/templates/deployment.yaml]: the `| join ","` did the trick

**Review by cwrau — APPROVED** [review]

**Pull request merged**